frontend/src/components/dashboard/Dashboard.js

```diff
@@ -161,16 +164,24 @@ const DropDownList = ({ buttonRef, handleOpen, handleClose, items, open }) => {
   const classes = useStyles();
   const { startLoading } = useContext(UserContext);
 
-  const handleClick = () => {
-    startLoading();
+  const handleClick = (onClick) => {
+    if (onClick) {
+      onClick();
+    } else {
+      startLoading();
+    }
   };
 
   return (
     <Popper open={open} anchorEl={buttonRef.current}>
       <Paper onMouseEnter={handleOpen} onMouseLeave={handleClose} className={classes.menu}>
         <MenuList>
           {items?.map((item) => (
-            <Link key={item.url_slug} href={item.url_slug} onClick={handleClick}>
+            <Link
+              key={item.url_slug}
+              href={item.url_slug}
+              onClick={() => handleClick(item.onClick)}
```

Comment:
Is there a benefit here to using inline arrow functions over the bound handler?

Reply:
I am not sure whether I understand what you mean but `onClick` requires `Function` type and doing `onClick={() => handleClick(item.onClick)}` would just call `handleClick(item.onClick)` during rendering and insert the output as the parameter. I could insert `onClick={handleClick}` without the arrow but then I would not be able to pass the `item.onClick` parameter. Do you mean some other way? :-)

Reply:
Nope that makes sense! Thanks for explaining :-)